zebra/rt_netlink.c

```diff
@@ -1290,65 +1201,37 @@ static int netlink_route_change_read_unicast_internal(struct nlmsghdr *h,
 				}
 			}
 		}
-		if (nhe_id || ng) {
-			rib_add_multipath(afi, SAFI_UNICAST, &p,
-					  (struct prefix_ipv6 *)&src_p,
-					  re, ng, startup);
-			if (ng)
-				nexthop_group_delete(&ng);
-		} else {
-			/*
-			 * I really don't see how this is possible
-			 * but since we are testing for it let's
-			 * let the end user know why the route
-			 * that was just received was swallowed
-			 * up and forgotten
-			 */
-			zlog_err(
-				"%s: %pFX multipath RTM_NEWROUTE has a invalid nexthop group from the kernel",
-				__func__, &p);
-			zebra_rib_route_entry_free(re);
+		dplane_ctx_route_set_is_multipath(ctx, is_multipath);
+		dplane_ctx_route_set_nexthop_group(ctx, ng);
+
+		if (ng) {
+			nexthop_group_delete(&ng);
+			ng = NULL;
 		}
-	} else {
-		if (nhe_id) {
-			rib_delete(afi, SAFI_UNICAST, vrf_id, proto, 0, flags,
-				   &p, (struct prefix_ipv6 *)&src_p, NULL,
-				   nhe_id, table, metric, distance, true);
-		} else {
+	} else if (h->nlmsg_type == RTM_DELROUTE) {
+		dplane_ctx_set_op(ctx, DPLANE_OP_ROUTE_DELETE);
+		if (!nhe_id) {
 			if (!tb[RTA_MULTIPATH]) {
 				struct nexthop nh;
 
-				nh = parse_nexthop_unicast(
-					ns_id, rtm, tb, bh_type, index, prefsrc,
-					gate, afi, vrf_id);
-				rib_delete(afi, SAFI_UNICAST, vrf_id, proto, 0,
-					   flags, &p,
-					   (struct prefix_ipv6 *)&src_p, &nh, 0,
-					   table, metric, distance, true);
+				nh = parse_nexthop_unicast(ns_id, rtm, tb, bh_type, index, prefsrc,
+							   gate, afi);
 
+				dplane_ctx_route_set_nexthop(ctx, &nh);
 				if (nh.nh_label)
 					nexthop_del_labels(&nh);
 
 				if (nh.nh_srv6)
 					nexthop_del_srv6_seg6(&nh);
-			} else {
-				/* XXX: need to compare the entire list of
-				 * nexthops here for NLM_F_APPEND stupidity */
-				rib_delete(afi, SAFI_UNICAST, vrf_id, proto, 0,
-					   flags, &p,
-					   (struct prefix_ipv6 *)&src_p, NULL, 0,
-					   table, metric, distance, true);
 			}
 		}
 	}
 
-	ret = 1;
+	dplane_ctx_set_notif_provider(ctx, notif_src);
```

Comment:
I don't really follow this, and there's no comment. why - what does this mean, who would use it and how?